lntest/itest/lnd_macaroons_test.go

```diff
@@ -374,9 +372,7 @@ func testBakeMacaroon(net *lntest.NetworkHarness, t *harnessTest) {
 
 	for _, tc := range testCases {
 		tc := tc
-		t.t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
-
+		t.t.Run(tc.name, func(tt *testing.T) {
```

Comment:
what would crash?

Reply:
Without the crash fix commit, the crash always happens here: 

```[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xe7769c]

goroutine 1048 [running]:
testing.tRunner.func1(0xc0005f8a00)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0xfe75c0, 0x1cd1c10)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/lightningnetwork/lnd/lntest/itest.testMacaroonAuthentication.func7(0x1477700, 0xc00038a060, 0xc0005f8a00)
	/home/bhandras/work/lnd/lntest/itest/lnd_macaroons_test.go:162 +0x1ac
github.com/lightningnetwork/lnd/lntest/itest.testMacaroonAuthentication.func8(0xc0005f8a00)
	/home/bhandras/work/lnd/lntest/itest/lnd_macaroons_test.go:208 +0xd7
testing.tRunner(0xc0005f8a00, 0xc00060f920)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350```

Reply:
Do we know why? I'm trying to understand what changed in this PR that causes it to crash.

Reply:
The renaming is just to not get shadowed variables. The main change is that the test used `t.Parallel()` and now doesn't.
It was caused by mixing serial and parallel code which caused the nodes to be cleaned up too soon.
Simplified it looked like this:

```
t.Run("main test case loop" func(t *testing.T) {
    SetupNodes()
    runTestCase()
    TeardownNodes()
})
```

But because in this case `runTestCase()` spawned a bunch of parallel tests, the function returned immediately and `TeardownNodes()` was called while they were still running.

Introducing an additional layer of serial test closures would also have fixed the problem:
```
t.Run("main test case loop", func(t *testing.T) {
    SetupNodes()
    t.Run("sub test case", func(t *testing.T) {
        runTestCase()
    })
    TeardownNodes()
})
```

But because those were very short-running tests, it didn't really matter to serialize them.

Reply:
Thanks @guggero  for the explanation!